third-party/cmake/BuildLuarocks.cmake

```diff
@@ -80,24 +129,26 @@ if(USE_BUNDLED_BUSTED)
     ARGS build luasocket 3.0rc1-2 ${LUAROCKS_BUILDARGS}
     COMMAND ${LUAROCKS_BINARY}
     ARGS build xml 1.1.1-1 ${LUAROCKS_BUILDARGS}
-    COMMAND touch ${DEPS_LIB_DIR}/luarocks/rocks/stable-busted-deps
+    COMMAND touch ${HOSTDEPS_LIB_DIR}/luarocks/rocks/stable-busted-deps
     DEPENDS lpeg)
   add_custom_target(stable-busted-deps
-    DEPENDS ${DEPS_LIB_DIR}/luarocks/rocks/stable-busted-deps)
+    DEPENDS ${HOSTDEPS_LIB_DIR}/luarocks/rocks/stable-busted-deps)
   
-  add_custom_command(OUTPUT ${DEPS_BIN_DIR}/busted
+  add_custom_command(OUTPUT ${HOSTDEPS_BIN_DIR}/busted
     COMMAND ${LUAROCKS_BINARY}
     ARGS build https://raw.githubusercontent.com/Olivine-Labs/busted/v2.0.rc8-0/busted-2.0.rc8-0.rockspec ${LUAROCKS_BUILDARGS}
     DEPENDS stable-busted-deps)
   add_custom_target(busted
-    DEPENDS ${DEPS_BIN_DIR}/busted)
+    DEPENDS ${HOSTDEPS_BIN_DIR}/busted)
   
-  add_custom_command(OUTPUT ${DEPS_LIB_DIR}/luarocks/rocks/nvim-client
+  add_custom_command(OUTPUT ${HOSTDEPS_LIB_DIR}/luarocks/rocks/nvim-client
     COMMAND ${LUAROCKS_BINARY}
-    ARGS build https://raw.githubusercontent.com/neovim/lua-client/8cc5b6090ac61cd0bba53ba984f15792fbb64573/nvim-client-0.0.1-11.rockspec ${LUAROCKS_BUILDARGS} LIBUV_DIR=${DEPS_INSTALL_DIR}
+    ARGS build https://raw.githubusercontent.com/neovim/lua-client/8cc5b6090ac61cd0bba53ba984f15792fbb64573/nvim-client-0.0.1-11.rockspec ${LUAROCKS_BUILDARGS} LIBUV_DIR=${HOSTDEPS_INSTALL_DIR}
     DEPENDS busted libuv)
   add_custom_target(nvim-client
-    DEPENDS ${DEPS_LIB_DIR}/luarocks/rocks/nvim-client)
+    DEPENDS ${HOSTDEPS_LIB_DIR}/luarocks/rocks/nvim-client)
+  list(APPEND THIRD_PARTY_DEPS nvim-client)
   
-  list(APPEND THIRD_PARTY_DEPS stable-busted-deps busted nvim-client)
+  list(APPEND THIRD_PARTY_DEPS stable-busted-deps busted)
```

Comment:
Why shuffle the `nvim-client` dependency into it's own list-append?  Is there an ordering issue?